test/renderers/table-array.control.test.tsx

```diff
@@ -106,260 +112,429 @@ test('render two children', t => {
+     <TableArrayControl schema={t.context.schema}
+   uischema={t.context.uischema}
+   visible={false}
+     />
+     </Provider>
+ );
+   const control = findRenderedDOMElementWithClass(tree, 'control') as HTMLElement;
+   t.true(control.hidden);
+ });
+
+test('show by default', t => {
+   const store = initJsonFormsStore(
+     t.context.data,
+     t.context.schema,
+     t.context.uischema
+   );
+   const tree = renderIntoDocument(
+     <Provider store={store}>
+     <TableArrayControl schema={t.context.schema}
+   uischema={t.context.uischema}
+ />
+   </Provider>
+ );
+   const control = findRenderedDOMElementWithClass(tree, 'control') as HTMLElement;
+   t.false(control.hidden);
+ });
+
+test('single error', t => {
+   const store = initJsonFormsStore(t.context.data, t.context.schema, t.context.uischema);
+   const tree = renderIntoDocument(
+     <Provider store={store}>
+       <TableArrayControl schema={t.context.schema}
+                       uischema={t.context.uischema}
+       />
+     </Provider>
+   );
+   const validation = findRenderedDOMElementWithClass(tree, 'validation');
+   store.dispatch(update('test', () => 2));
+   store.dispatch(validate());
+   t.is(validation.textContent, 'should be array');
+ });
+
+test('multiple errors', t => {
+   const store = initJsonFormsStore(t.context.data, t.context.schema, t.context.uischema);
+   const tree = renderIntoDocument(
+     <Provider store={store}>
+     <TableArrayControl schema={t.context.schema}
+                     uischema={t.context.uischema}
+
+     />
+     </Provider>
+   );
+   const validation = findRenderedDOMElementWithClass(tree, 'validation');
+   store.dispatch(update('test', () => 3));
+   store.dispatch(validate());
+   t.is(validation.textContent, 'should be array');
+ });
+
+test('empty errors by default', t => {
+   const store = initJsonFormsStore(
+     t.context.data,
+     t.context.schema,
+     t.context.uischema
+   );
+   const tree = renderIntoDocument(
+     <Provider store={store}>
+     <TableArrayControl schema={t.context.schema}
+   uischema={t.context.uischema}
+
+ />
+   </Provider>
+ );
+   const validation = findRenderedDOMElementWithClass(tree, 'validation');
+   t.is(validation.textContent, '');
+ });
+
+test('reset validation message', t => {
+   const store = initJsonFormsStore(
+     t.context.data,
+     t.context.schema,
+     t.context.uischema
+   );
+   const tree = renderIntoDocument(
+     <Provider store={store}>
+     <TableArrayControl schema={t.context.schema}
+   uischema={t.context.uischema}
+
+ />
+   </Provider>
+ );
+   const validation = findRenderedDOMElementWithClass(tree, 'validation');
+   store.dispatch(update('test', () => 3));
+   store.dispatch(validate());
+   t.is(validation.textContent, 'should be array');
+   store.dispatch(update('test', () => []));
+   store.dispatch(validate());
+   t.is(validation.textContent, '');
+ });
+
+/*test('validation of nested schema', t => {
```

Comment:
Why is this commented? Maybe rather mark it as `failing` or `skip` or remove it completely?